### PR TITLE
Bank lore search

### PR DESCRIPTION
**Pull request by ryanzhoudev**
Allows users to search item lores instead of item names in the bank search bar.
Format is 
`lore:search_term`
eg.
`lore:mana regen`
would return items on that page with mana regen.

### CODE REVIEW

File: src/main/java/com/wynntils/modules/utilities/overlays/inventories/BankOverlay.java [hunk 1]
```diff
@@ -393,11 +393,24 @@ private void searchPageForItems(ChestReplacer bankGui) {
...
+                    searchedItems.add(is);
+            }
+        } else {
+            for (int i = 0; i < bankGui.getLowerInv().getSizeInventory(); i++) {
+                if (i % 9 > 6) continue; // ignore sidebar items
 
-            ItemStack is = bankGui.getLowerInv().getStackInSlot(i);
-            if (TextFormatting.getTextWithoutFormattingCodes(is.getDisplayName()).toLowerCase().contains(searchText)) searchedItems.add(is);
+                ItemStack is = bankGui.getLowerInv().getStackInSlot(i);
+                if (TextFormatting.getTextWithoutFormattingCodes(is.getDisplayName()).toLowerCase().contains(searchText))
```
Comment (magicus) [inline src/main/java/com/wynntils/modules/utilities/overlays/inventories/BankOverlay.java]: Please use { }. If statements without blocks are super duper error prone! 

The simple one-line "if ... return" that are used liberally in Wynntils are kind of okay, but not complex stuff like this. (And the other case above)

**Comment by magicus** [comment]
This fix screams for generalisation. :-) I won't force you, but I do think it'd be better off if you introduced a method like `getSearchString(ItemStack is, boolean isLore)` or something like that. Then read and strip "lore": and use it to set isLore. And just let this function return `ItemUtils.getStringLore(is))` or `is.getDescription()` depending on the value.

**Comment by Incompleteusern** [comment]
LGTM. As magicus said, generalization is possible.

EDIT: Combination with ItemsPage is implausible

**Comment by ryanzhoudev** [comment]
> This fix screams for generalisation. :-) I won't force you, but I do think it'd be better off if you introduced a method like `getSearchString(ItemStack is, boolean isLore)` or something like that. Then read and strip "lore": and use it to set isLore. And just let this function return `ItemUtils.getStringLore(is))` or `is.getDescription()` depending on the value.

Bit confused on what you mean by this, I'm gonna write something up and you can see if it's what you meant?

File: src/main/java/com/wynntils/modules/utilities/overlays/inventories/BankOverlay.java [hunk 2]
```diff
@@ -393,14 +392,31 @@ private void searchPageForItems(ChestReplacer bankGui) {
...
+            if (getSearchString(is, searchLore).contains(searchText)) {
+                searchedItems.add(is);
+            }
+
         }
     }
 
+    private String getSearchString(ItemStack is, boolean isLore) {
+        if (isLore) return TextFormatting.getTextWithoutFormattingCodes(ItemUtils.getStringLore(is).toLowerCase());
+        return TextFormatting.getTextWithoutFormattingCodes(is.getDisplayName().toLowerCase());
```
Comment (magicus) [inline src/main/java/com/wynntils/modules/utilities/overlays/inventories/BankOverlay.java]:
```suggestion
        if (isLore) {
          return TextFormatting.getTextWithoutFormattingCodes(ItemUtils.getStringLore(is).toLowerCase());
        } else {
          return TextFormatting.getTextWithoutFormattingCodes(is.getDisplayName().toLowerCase());
        }
```

**Review by magicus — APPROVED** [review]
Yes, exactly. Much better! I added a suggestion with code formatting. Apart from that, this looks good to me now.

File: src/main/java/com/wynntils/modules/utilities/overlays/inventories/BankOverlay.java (hunk 1 above)
Comment (ryanzhoudev) [reply src/main/java/com/wynntils/modules/utilities/overlays/inventories/BankOverlay.java]: Got it, thanks!

**Pull request merged**